Odata-docs/webapi-8/entityset-routing.md

```diff
@@ -0,0 +1,523 @@
+```json
+{
+    "@odata.context": "http://localhost:5000/odata/$metadata#Shapes/$entity",
+    "Id": 4,
+    "Area": 36.0
+}
+```
+
+If you look through the response headers, you should find the location (resource path) of the created entity: http://localhost:5000/odata/Shapes(4). Querying the `Shapes` entity set should confirm that the entity was successfully added.
+
+### Adding a derived entity to an entity set
+To create a derived entity in an entity set, the client sends a `POST` request to that entity set's URL with the fully-qualified name of the derived type appended at the end - `/cast` (cast from type casting).
+
+The route template for this request is: `POST ~/{entityset}/{cast}`
+
+The following `POST` request adds a `Circle` entity to the `Shapes` entity set:
+```http
+POST http://localhost:5000/odata/Shapes/EntitySetRouting.Models.Circle
+```
+
+Here's the request body:
+```json
+{
+    "Id": 5,
+    "Radius": 1.4,
+    "Area": 6.16
+}
+```
+
+For the above request to be conventionally-routed, a controller action named `PostFromCircle` (or `PostShapeFromCircle`) is expected:
+```csharp
+public ActionResult PostFromCircle([FromBody] Circle circle)
+{
+    shapes.Add(circle);
+
+    return Created(circle);
+}
+```
+
+The response status code should be `201`. The following JSON payload shows the expected response:
+```json
+{
+    "@odata.context": "http://localhost:5000/odata/$metadata#Shapes/EntitySetRouting.Models.Circle/$entity",
+    "Id": 5,
+    "Area": 6.16,
+    "Radius": 1.4
+}
+```
+
+If you look through the response headers, you should find the location (resource path) of the created derived entity: http://localhost:5000/odata/Shapes(5). Querying the `Shapes` entity set should confirm that the derived entity was successfully added.
+
+### Patching a collection of entities
+The semantics of `PATCH` is to merge the content in the request payload with the entity's or entities' current state, applying the update only to those components specified in the request body. To patch a collection of entities in an entity set, the client sends a `PATCH` request to that entity set's URL.
+
+The route template for this request is: `PATCH ~/{entityset}`
+
+The following `PATCH` request patches shape 1 (a circle) and shape 2 (a rectangle):
+```http
+PATCH http://localhost:5000/odata/Shapes
+```
+
+Here's the request body:
+```json
+{
+    "value": [
+        {
+            "@odata.type": "#EntitySetRouting.Models.Circle",
+            "Id": 2,
+            "Radius": 0.7,
+            "Area": 1.54
+        },
+        {
+            "@odata.type": "#EntitySetRouting.Models.Rectangle",
+            "Id": 3,
+            "Length": 8,
+            "Width": 4,
+            "Area": 32
+        }
+    ]
+}
+```
+
+For the above request to be conventionally-routed, a controller action named `Patch` (or `PatchShape`) is expected:
+```csharp
+public ActionResult Patch([FromBody] DeltaSet<Shape> deltaSet)
```

Comment:
this is great advertisement of the deltaset

Reply:
should the request body contain the @odata.context in 8.x?

Reply:
@ElizabethOkerio Please note that the `DeltaSet<T>` in 8.0 only serves to allow someone to send a collection of `Delta<T>`. Support for bulk operations is not implemented. It therefore doesn't expect the `@odata.context` property in the payload